pkg/controllers/vrep/replication_status_reconciler.go

```
@@ -0,0 +1,279 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	statusStarted   = "started"
+	statusFailed    = "failed"
+	statusCompleted = "completed"
+
+	opLoadSnapshotPrep = "load_snapshot_prep"
+	opDataTransfer     = "data_transfer"
+	opLoadSnapshot     = "load_snapshot"
+)
+
+type ReplicationStatusReconciler struct {
+	client.Client
+	VRec         *VerticaReplicatorReconciler
+	Vrep         *v1beta1.VerticaReplicator
+	dispatcher   vadmin.Dispatcher
+	SourcePFacts *podfacts.PodFacts
+	TargetPFacts *podfacts.PodFacts
+	Log          logr.Logger
+	TargetInfo   *ReplicationInfo
+}
+
+func MakeReplicationStatusReconciler(cli client.Client, r *VerticaReplicatorReconciler, vrep *v1beta1.VerticaReplicator,
+	log logr.Logger) controllers.ReconcileActor {
+	return &ReplicationStatusReconciler{
+		Client:     cli,
+		VRec:       r,
+		Vrep:       vrep,
+		Log:        log.WithName("ReplicationStatusReconciler"),
+		TargetInfo: &ReplicationInfo{},
+	}
+}
+
+func (r *ReplicationStatusReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	// no-op if replication is not being done asynchronously
+	if r.Vrep.Spec.Mode != replicationModeAsync {
+		r.Log.Info("Stopped reconciling status: not async")
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if ReplicationComplete is present (either true or false)
+	isPresent := r.Vrep.IsStatusConditionPresent(v1beta1.ReplicationComplete)
+	if isPresent {
+		r.Log.Info("Stopped reconciling status: complete")
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if ReplicationReady is false
+	isFalse := r.Vrep.IsStatusConditionFalse(v1beta1.ReplicationReady)
+	if isFalse {
+		r.Log.Info("Stopped reconciling status: not ready")
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if there is no transaction ID we can use to query replication status
+	if r.Vrep.Status.TransactionID == 0 {
+		r.Log.Info("Stopped reconciling status: transaction ID 0")
```

We should add one more check to see if replicating is started, if it is, we should return an error saying "cannot find transcation ID".